lib/odoo.js

```diff
@@ -38,11 +38,17 @@ class Odoo {
 		};
 
 		return got.post(`${config.host}:${config.port}/web/session/authenticate`, {json: true, headers, body})
-			.then(res => new OdooClient(res.body.result, {
-				host: config.host,
-				port: config.port,
-				sid: res.headers['set-cookie'][0].split(';')[0]
-			}));
+			.then(res => {
+				if (res.body.error) {
+					// HTTP Status is 200 / OK!
+					throw new Error(res.body.error.message);
```

Comment:
I agree, you should always reject with an error. But isn't `res.body.error` an Error object? It's definitely not a string.